### DISCUSSION TITLE
Unable to use OpenImageIO port in CMake project

### DISCUSSION THREAD

**Discussion by daniilvinn** [microsoft/vcpkg#29106] [Q&A] [answered]
Hi guys! I tried to add OpenImageIO port to my CMake project. I'm pretty sure it is not a bug, but still would like to get an answer on my question.

The problem is that it can't locate ZLIB and PNG (which is basically libpng), even though they ARE installed (OpenImageIO installation did it recursively, since it depends on it).

The way I run my cmake build:
`call cmake ../ -DCMAKE_TOOLCHAIN_FILE=C:/local/vcpkg/scripts/buildsystems/vcpkg.cmake`

My root CMakeLists.txt:
```cmake_minimum_required(VERSION 3.10)
set (CMAKE_SYSTEM_VERSION 10.0.22621.0)

project(Motus3D)

include("dependencies.cmake")

find_package(OpenImageIO CONFIG REQUIRED)

set(CMAKE_CXX_STANDARD 20)
set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/obj/)
set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/obj/)
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin/)

add_subdirectory(Motus3D)
add_subdirectory(Visus)
add_subdirectory(Sandbox2D)
add_subdirectory(Scene)
add_subdirectory("Motus3D/thirdparty/GLFW")
add_subdirectory("Motus3D/thirdparty/robin-hood-hashing")
add_subdirectory("Motus3D/thirdparty/Assimp")
add_subdirectory("Motus3D/thirdparty/imgui")

set(CMAKE_DEFAULT_STARTUP_PROJECT Sandbox)
```

What I get in console after running `cmake_build.bat` (my own file to build):

```E:\dev\Motus-3D\build>call cmake ../ -DCMAKE_TOOLCHAIN_FILE=C:/local/vcpkg/scripts/buildsystems/vcpkg.cmake
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)
CMake Error at C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.25/Modules/FindPNG.cmake:159 (find_package_handle_standard_args)
  C:/Program Files/CMake/share/cmake-3.25/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  C:/local/vcpkg/installed/x64-windows/share/openimageio/OpenImageIOConfig.cmake:31 (find_dependency)
  CMakeLists.txt:8 (find_package)


-- Configuring incomplete, errors occurred!
See also "E:/dev/Motus-3D/build/CMakeFiles/CMakeOutput.log".
See also "E:/dev/Motus-3D/build/CMakeFiles/CMakeError.log".

E:\dev\Motus-3D\build>PAUSE
```

Also I can provide any other information you need to solve this issue.
Struggling for like 2 days just to install 1 library... Hope someone helps!

**Comment by dg0yt** [comment, marked as answer]
According to the call stack, you don't use vcpkg's cmake integration (i.e. the toolchain file) and that's why you miss the fixup added by vcpkg's special find_package wrappers.

**Reply by dg0yt** [reply]
Note: The toolchain must be set the first time you run CMake for a build dir. You cannot add it later.

**Reply by daniilvinn** [reply]
What do you mean by "set the first time you run CMake for a build dir"? Do I need to delete `build/` dir and then run it again?

**Reply by daniilvinn** [reply]
Yep, I just had to delete CMakeCache and it fixed the problem. Thanks!

**Comment by dg0yt** [comment]
You should also add `cmake_minimum_required(VERSION 3.25)` or similar before the `project` line.

**Reply by daniilvinn** [reply]
Just did it, no changes. Still can't locate zlib and libpng.
